### PR TITLE
Assistant Builder action screen take II

### LINKED ISSUE #3783: [Meru - AB] Layout - New Actions screen
Details on what to do in the **[Figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4900-39294&mode=design&t=6tNvt0C7wUTsOHc4-4)** (also, see how it works in the [Prototype](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?page-id=3366%3A31345&type=design&node-id=4900-40669&viewport=1173%2C2546%2C0.27&t=lV2M2VcNHLWO3Bpe-1&scaling=min-zoom&starting-point-node-id=4900%3A40669) + context & discussion in the [making issue](https://github.com/dust-tt/decisions/issues/164))

- Action selection is now multi-steps (see prototype). First choose action, then if 'use data source' choose how to use datasource (then pick datasource); for dust app just choose the app
- Note the various [action management phases](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4758-31108&mode=design&t=6tNvt0C7wUTsOHc4-4) in the figma
- "Help Me" button is out of scope of this issue, will be done in another one
- New "Select data source" will also be done in another issue

### PR DESCRIPTION
**Pull request by spolu**
## Description

Closes: https://github.com/dust-tt/dust/issues/3783

- Finish applying the design changes
- Updates the ContentMessage when no data source are visible + hides components as needed
- Remove most description text (TBD if we want to re-introduce it with a side modal)
- Add ContentMessage for Dust Apps (when there is none)

Screenshot from 2024-02-26 11-52-28
Screenshot from 2024-02-26 11-52-33
Screenshot from 2024-02-26 11-52-48
Screenshot from 2024-02-26 11-52-54
Screenshot from 2024-02-26 11-52-23

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/components/assistant_builder/ActionScreen.tsx
Comment (philipperolet) [inline front/components/assistant_builder/ActionScreen.tsx:376]: nit: copy change. feels like saying to the user "You don't have data sources, and you can't do anything about it, cheers"

I'd phrase it "Contact a Dust Admin to add connections, URLs or folders". (Same for builder)

Comment (philipperolet) [reply front/components/assistant_builder/ActionScreen.tsx:376]: just seeing this copy is from former code; but good occasion for a quick improvement IMHO

File: front/components/assistant_builder/ActionScreen.tsx
Comment (philipperolet) [inline front/components/assistant_builder/ActionScreen.tsx:557]: Copy above is not in the figma. I think it's worth keeping it :) but just mentioning

File: front/components/assistant_builder/ActionScreen.tsx
Comment (philipperolet) [inline front/components/assistant_builder/ActionScreen.tsx:602]: Similar to copy comment above
"Contact an Admin or Builder if you need a Dust Application."

File: front/components/assistant_builder/ActionScreen.tsx
Comment (philipperolet) [inline front/components/assistant_builder/ActionScreen.tsx:630]: Not in figma; should we keep it?

File: front/components/assistant_builder/AssistantBuilder.tsx
Comment (philipperolet) [inline front/components/assistant_builder/AssistantBuilder.tsx:1038]: IMHO let's kill the code rather than add this, we risk forgetting it

We can't merge before the slack integration in dropdown is merged anyways, and once it is then the code should be removed

File: front/components/assistant_builder/DataSourceSelectionSection.tsx
Comment (philipperolet) [inline front/components/assistant_builder/DataSourceSelectionSection.tsx:14]: nit: relative import

File: front/components/EmptyCallToAction.tsx
Comment (philipperolet) [inline front/components/EmptyCallToAction.tsx:17]: I see you pass `icon={null}` below rather than omitting the prop

IMHO keeping the inteface `icon?: ComponentType` was fine. Second choice would be `icon: Comp | null` . I prefer the 1st because it's already the most used in code and IMHO nicer to read than having `icon={null}`, but won't fight for it.

However allowing both undefined and null seems weird, would choose one or the other

**Review by philipperolet — CHANGES_REQUESTED** [review]
Looks great ! Left a few comments, nothing major

File: front/components/assistant_builder/ActionScreen.tsx
Comment (spolu) [reply front/components/assistant_builder/ActionScreen.tsx:630]: Nothing makes it impossible for anyone to even find a place to read about this. We need to come up with a documentation + link it from here.

We'll likely do the same for Query Tables (at least that's the plan as per the project issues).

Not doing it here but def TODO, but removing everything means there's no way to learn about his at all..,

File: front/components/assistant_builder/AssistantBuilder.tsx
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:1038]: @lasryaric is working on it killing will make his life harder

File: front/components/assistant_builder/ActionScreen.tsx
Comment (spolu) [reply front/components/assistant_builder/ActionScreen.tsx]: tweaked

File: front/components/assistant_builder/DataSourceSelectionSection.tsx
Comment (spolu) [reply front/components/assistant_builder/DataSourceSelectionSection.tsx]: :+1:

**Pull request merged**
